lib/prefetch.js

```diff
@@ -110,7 +110,7 @@ if (hasServiceWorkerSupport()) {
 
 function getPrefetchUrl (href) {
   let { pathname } = urlParse(href)
-  const url = `/_next/${__NEXT_DATA__.buildId}/pages${pathname}`
+  const url = `/_next/${__NEXT_DATA__.buildId}/${__NEXT_DATA__.pagesDirectory}${pathname}`
```

Comment:
I don't *think* keeping `./.next/pages` regardless would work with SSR if you had a deeper-nested custom pages folder (ala `views/pages`, see discussion [here](https://github.com/zeit/next.js/pull/936#issuecomment-276404581))